features/viewfolder/impl/src/main/kotlin/io/element/android/features/viewfolder/impl/file/ViewFileState.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.viewfolder.impl.file
+
+import kotlinx.collections.immutable.ImmutableList
+
+data class ViewFileState(
+    val name: String,
+    val lines: ImmutableList<String>,
```

Comment:
Maybe this should be an `AsyncData` and the UI should have a loader? For large files, the viewer will display an 'empty file' text for 1-2s before actually displaying the file contents.

Reply:
Yes, you're right, I've been a bit lazy here :). 4601e2acd33c67bd60920cb4b9351946941dba8d